doc/docs/learn/create-an-app-with-monomer.md

```diff
@@ -0,0 +1,568 @@
+```
+
+### Finalizing our Changes
+Near the bottom of the `NewChainApp` function, comment out the following lines:
+```go
+// app/app.go
+
+/*protoFiles, err := proto.MergedRegistry()
+if err != nil {
+    panic(err)
+}
+err = msgservice.ValidateProtoAnnotations(protoFiles)
+if err != nil {
+    // Once we switch to using protoreflect-based antehandlers, we might
+    // want to panic here instead of logging a warning.
+    _, _ = fmt.Fprintln(os.Stderr, err.Error())
+}*/
+```
+and remove this import: `github.com/cosmos/cosmos-sdk/types/msgservice`. 
+
+
+
+That's it for `app.go`! Next we'll add a CLI command to interact with Monomer.
+
+### Defining the Monomer Command
+In `cmd/rolld/commands.go`, import `github.com/polymerdao/monomer/integrations` and add the following block to the `initRootCmd`
+function. This defines the `monomer` command which utilizes our custom start
+command hook that handles the integration between Monomer and the Cosmos SDK:
+```go
+// cmd/rolld/commands.go
+
+import (
+    // ...
+    "github.com/polymerdao/monomer/integrations"
+    // ...
+)
+
+func initRootCmd(
+  rootCmd *cobra.Command,
+  txConfig client.TxConfig,
+  interfaceRegistry codectypes.InterfaceRegistry,
+  appCodec codec.Codec,
+  basicManager module.BasicManager,
+) {
+  // ...
+	
+  // Remove or comment out the following line:
+  /* server.AddCommands(rootCmd, app.DefaultNodeHome, newApp, appExport, addModuleInitFlags) */
+
+  // --- Add this block ---
+  monomerCmd := &cobra.Command{
+	Use:     "monomer",
+	Aliases: []string{"comet", "cometbft", "tendermint"},
+	Short:   "Monomer subcommands",
+  }
+  monomerCmd.AddCommand(server.StartCmdWithOptions(newApp, app.DefaultNodeHome, server.StartCmdOptions{
+    StartCommandHandler: integrations.StartCommandHandler,
+  }))
+  rootCmd.AddCommand(monomerCmd)
+  // --- End block ---
+  
+  // ...
+}
+```
+
+## Building the Application
+Before we can build our application, we need to add the following replace directive in `go.mod` and then run `go mod
+tidy`: 
+```go
+github.com/ethereum/go-ethereum => github.com/joshklop/op-geth v0.0.0-20240515205036-e3b990384a74
+```
+
+Now we can build our application by running 
+```bash
+make install 
+```
+from the root of the
+`rollchain/` directory. This will install our chain binary, `rolld`, to the
+`$GOBIN` directory.
+
+## Configuring the Application 
+
+There's one last step before we can run our application: We need to write
+its genesis file. We can generate one by running
+```bash
+rolld init my-monomer-app --chain-id=1
+```
+
+:::important
+Make sure to use a numeric chain ID. Monomer will fail to start if the chain ID
+cannot be parsed as a `uint64`.
+:::
+
+This will create a new directory in `~/.rollchain/config/` with the necessary
+application configuration files. For the sake of this tutorial, we will tweak
+the generated `genesis.json` file slightly. Copy the following genesis file to
+`~/.rollchain/config/genesis.json`.
```

Comment:
We should add reasoning for tweaking it. How does this differ from the genesis file created on `init` above?

Reply:
Essentially the only differences between the generated genesis file and our tweaked version is using a numeric value for `chain_id` and including the `testmodule` module. Will add a line to make this clear

Reply:
Something like this maybe?

---

This will create a new directory in `~/.rollchain/config` with the necessary application configuration files. For the sake of this tutorial, we **must tweak the generated `genesis.json` file** slightly.
1. Update the `chain_id` parameter to a numeric value.
2. Include `x/testmodule` in `app_state` so it can be loaded at genesis.

Copy the following genesis file to `~/.rollchain/config/genesis.json` or make the changes directly.

Reply:
I pushed the above ^. Lmk if this is good to resolve

Reply:
Looks good, thanks!